baseplate/experiments/__init__.py

```diff
@@ -23,23 +23,26 @@ class ExperimentsContextFactory(ContextFactory):
     This factory will attach a new :py:class:`baseplate.experiments.Experiments`
     to an attribute on the :term:`context object`.
     """
-    def __init__(self, path, event_queue):
+    def __init__(self, path, event_queue, event_constructor=None):
         """ExperimentContextFactory constructor
 
         :param str path: Path to the experiment config file.
         :param baseplate.events.EventQueue event_queue: Event queue used for
             logging bucketing events to the event pipeline.  You can set this
-            to None to disable bucketing events entirely.
+            to None to disable bucketing events entirely. This queue should be
+            configured to use baseplate.events.serialize_v2_event
         """
         self._filewatcher = FileWatcher(path, json.load)
         self._event_queue = event_queue
+        self._event_constructor = event_constructor
```

Comment:
This pattern of passing the constructor in seems pretty inelegant to me especially since you have to set it up in the exact format that you expect in Baseplate.  

What do you think about instead passing making an `EventLogger` interface that just has one method `def log(self, event_name, **inputs)`.  You can make a "debug" logger here that will just log everything to `logging.debug` but you could also make one that fires v2 events and give that to the `Experiments` client.  Then instead of calling `self._log_bucketing_event` you get rid of `_log_bucketing_event` and just call `self._event_logger.log('bucket', **inputs)`.